flight/Modules/Actuator/actuator.c

```diff
@@ -611,6 +648,80 @@ static void post_process_scale_and_commit(float *motor_vect,
 		ActuatorCommandGet(&command);
 	}
 
+#if defined(PIOS_INCLUDE_ESCTELEMETRY)
+	if (PIOS_ESCTelemetry_IsAvailable()) {
+
+		/* Process telemetry request stuff before servo settings. */
+
+		if (!tlm) {
+			tlm = PIOS_malloc_no_dma(sizeof(*tlm));
+			if (!tlm)
+				PIOS_Assert(0);
+
+			tlm->servo = 0xFF;
+			tlm->active = 0xFF;
+
+			ActuatorTelemetryGet(&tlm->telemetry);
+		}
+
+		/* Just cycle naively through all servos. */
+		if (tlm->servo >= MAX_MIX_ACTUATORS && PIOS_DELAY_GetuSSince(tlm->cycle_start) > ESC_TELEM_CYCLE_uS) {
+			tlm->servo = 0;
+			tlm->ok = 0;
+			tlm->cycle_start = PIOS_DELAY_GetuS();
+		}
+
+		if ((tlm->active == 0xFF) && (tlm->servo < MAX_MIX_ACTUATORS)) {
+			/* If active servo is 0xFF, no request is active, either because we're starting over,
+			   the servo doesn't exists, or isn't configured to DShot. */
+
+			tlm->active = tlm->servo;
+			tlm->servo++;
+
+			tlm->req_start = PIOS_DELAY_GetuS();
+
+			/* We don't care about ongoing requests, because we shouldn't even be here, if one is. */
+			if (PIOS_Servo_RequestTelemetry(tlm->active) < -1)
```

Comment:
What does -1 mean? I always prefer to use enums when actually differentiating errors. e.g.
`if (PIOS_*() == -EBUSY)` or similar is a lot more readable than `if (PIOS_*() == -5)`.